### PR TITLE
Add `doc` fields to script factory

### PR DESCRIPTION
**Pull request by nik9000**
This causes painless to generate the `docFields` method on every script
factory. It returns a `List<String>` containing all of the constant
keys passed to the `doc` variable's `get` method.

### CODE REVIEW

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-sample-windows

**Comment by nik9000** [comment]
@elasticmachine update branch

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessScriptEngine.java [hunk 1]
```diff
@@ -341,6 +341,24 @@ public Loader run() {
         deterAdapter.returnValue();
         deterAdapter.endMethod();
 
+        org.objectweb.asm.commons.Method docFields = new org.objectweb.asm.commons.Method(methodName,
```
Comment (jdconrad) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessScriptEngine.java:344]: We should probably only have this method if it's on the base factory.

**Review by jdconrad — APPROVED** [review]
LGTM. Thanks for adding this! One comment to address please.

**Comment by nik9000** [comment]
Thanks @jdconrad ! I've marked this one as "backport pending" because I'll need the doc fields work to be able to backport it. I'm quite happy to wait as long as it takes!

**Comment by jdconrad** [comment]
@nik9000 Understood about the backport. I'll put together some giant PR soon that will likely include this change as well.

**Comment by nik9000** [comment]
@jdconrad backported this with #61755.

**Pull request merged**